src/lib/config.js

```diff
@@ -116,13 +116,44 @@ function parseLedgers () {
   })
 }
 
+function parseRoutes () {
+  const routes = JSON.parse(Config.getEnv(envPrefix, 'ROUTES') || '[]')
+
+  for (let route of routes) {
+    if (
+      !route.connectorLedger.match(/^[a-zA-Z0-9._~-]+\.$/) ||
+      typeof route.connectorLedger !== 'string' ||
```

Comment:
You should check if its a string _before_ matching it against a regexp, not after.